test/windows/UnitTests.cpp

```diff
@@ -4346,6 +4349,67 @@ Error code: Wsl/Service/RegisterDistro/WSL_E_DISTRIBUTION_NAME_NEEDED\r\n";
             VERIFY_IS_FALSE(std::filesystem::exists(basePath));
         }
 
+        // Distribution with overridden default location but without explicit name (should use GUID)
+        {
+            auto cleanup = wil::scope_exit_log(
+                WI_DIAGNOSTICS_INFO, []() {
+                    auto distros = wsl::windows::common::SvcComm().EnumerateDistributions();
+                    for (const auto& distro : distros)
+                    {
+                        auto name = distro.DistributionName.get();
+                        if (wcsstr(name, L"Ubuntu") != nullptr)
+                        {
+                            std::wstring cmd = std::wstring(L"--unregister ") + name;
+                            LxsstuLaunchWsl(cmd.c_str());
+                            break;
+                        }
+                    }
+                });
+
+            auto currentPath = std::filesystem::current_path();
+            WslConfigChange wslconfig(std::format(L"[general]\ndistributionInstallPath = {}", EscapePath(currentPath.wstring())));
+
+            // Install without --name, so the distribution name will be auto-generated
+            InstallFromTar(g_testDistroPath.c_str(), L"");
+            
+            // Find the installed distribution
+            auto distros = wsl::windows::common::SvcComm().EnumerateDistributions();
+            std::wstring installedName;
+            for (const auto& distro : distros)

```

Comment:
We can remove this block, since `ValidateDistributionStarts` will validate that the distribution exists